src/app/pages/BridgeDepositPage/components/WalletSelector/index.tsx

```diff
@@ -64,6 +70,26 @@ export function WalletSelector() {
     chain,
   ]);
 
+  const changeNetwork = () => {
```

Comment:
Use `useCallback`. 

Also, I would check at the beginning if the network exists and if not, return. If it does exist, destructure the network object so you don't have to write e.g. `network?.explorer` but `explorer` instead.